### ISSUE TITLE
ecs-agent causes load spikes every 20 miutes

### ISSUE THREAD

**Issue by rhuddleston** [aws/amazon-ecs-agent#588]
![image](https://cloud.githubusercontent.com/assets/3074008/20027576/53b9bdae-a2dd-11e6-847b-cdd3c7d9e7e7.png)

Every one of our ecs nodes load average graph looked like the above spread over 6 clusters. Every 20 minutes there was a spike in load average. There was no spike in CPU activity, network, IO, memory etc. Basically no obvious cause. After extensive testing and trial and error I determined ecs agent itself was the cause of these spikes. Upgrading the agent did not fix the issue. Finally I fixed these spikes by added the following env before starting this container:

ECS_DISABLE_METRICS=true

This also stopped cloudwatch metrics per container but I was already getting a more detailed version of this with cadvisor and prometheus anyways. I thought I would add this bug in case anyone else runs across this and was scratching their heads. Also if anyone wants to track down the underlying issue I'd be very curios. This was running on ubuntu 16.04 using the official ecs-agent docker container.

**Comment by richardpen** [comment]
@rhuddleston

**Comment by richardpen** [comment]
Hi @rhuddleston,

Thanks for reporting this issue, IIUC the graph you provided shows the resource utilization of the node where ecs-agent is running, is that right? And could you tell us how this data was gathered so that we can reproduce on our side.

In order to help us debug the issue,,could you also provide us the metrics of ecs-agent during your test if you gathered with cadvisor, you can publish here or send to me at: penyin (at) amazon.com

Thanks,
Peng

**Comment by rhuddleston** [comment]
This is a simple one minute load average on the box. It was captured by node_exporter and prometheus but basically you can capture with almost anything including just running the "uptime" command and looking at the first value which corresponds to the one minute load average. With ECS_METRICS enabled the load spikes every 20 minutes but with it off it does not.

**Comment by aaithal** [comment]
Hi @rhuddleston , we did some benchmarking with cadvisor + node export + prometheus of the ECS Agent's CPU and Memory Utilization and found out that there were no untoward CPU and Memory Utilization spikes when running high cpu and memory utilization container workloads. We were launching 3 containers every minute on a single instance. The CPU load on a multi-core host can be tricky to interpret. If you find out there was a spike in utilization of CPU/Memory/Network (or any other resource), please let us know and we can investigate that more. Closing this for now. Feel free to reopen or create a new issue if you want to continue the conversation around resource utilization of the ECS Agent. Thanks!
